packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -456,6 +457,20 @@ class SliverGeometry {
   /// [SliverConstraints.remainingPaintExtent] provided during layout.
   final double scrollExtent;
 
+  /// The visual location of the first visible part of this sliver relative to
+  /// its layout position.
+  ///
+  /// For example, if the sliver wishes to paint visually before its layout
+  /// position, the [paintOrigin] is negative. The coordinate system this sliver
+  /// uses for painting is relative to this [paintOrigin].
+  ///
+  /// This value does not affect the layout of subsequent slivers. The next
+  /// sliver is still placed at [layoutExtent] after this sliver's layout
+  /// position. This value does affect where the [paintExtent] extent is
+  /// measured from when computing the [SliverConstraints.overlap] for the next
+  /// sliver.
```

Comment:
`Defaults to 0.0, i.e. by default slivers paint starting at their layout position.`